src/components/screens/Archive/ArchiveNoteInput.jsx

```diff
@@ -0,0 +1,202 @@
+import React, { useState } from "react";
+import { useArchiveContext } from "../../../global-context/archive-context";
+import { useGlobalVarContext } from "../../../global-context/global-variables";
+import { postArchivesService } from "../../../services/archive-services";
+
+// WORK IN PROGRESS
+const ArchiveNoteInput = ({ editObj, setEditObj }) => {
+  const { token } = useGlobalVarContext();
+  const [note, setNote] = useState(editObj.note);
+  const { isEditing, noteId } = editObj;
+  const { dispatch } = useArchiveContext();
+
+  const { title, content, stylePref, colorPref, isPinned } = note;
+  const editArchiveHandler = async (encToken, noteId, note) => {
+    console.log({ encToken, noteId, note });
+    const response = await postArchivesService(encToken, noteId, note);
+    console.log(response);
+    dispatch({ type: "ADD_ARCHIVE", payload: response.data.archives });
+  };
+
+  const changeStylePref = (pref) => {
+    setNote((prevObj) => {
+      const { stylePref } = prevObj;
+      if (/text-align/.test(pref)) {
```

Comment:
Got to know this new thing of testing it.Little confused on its implementation

Reply:
Those are there to test regex patterns in a given string 
we can have a discussion on this  in vc